drivers/mbox/mbox_ti_secproxy.c

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (c) 2025 Texas Instruments Incorporated.
+ *
+ * TI Secureproxy Mailbox driver for Zephyr's MBOX model.
+ */
+
+#include <zephyr/spinlock.h>
+#include <stdio.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/irq.h>
+#define LOG_LEVEL CONFIG_MBOX_LOG_LEVEL
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/mbox/mbox_ti_secproxy.h>
+LOG_MODULE_REGISTER(ti_secure_proxy);
+
+#define DT_DRV_COMPAT ti_secure_proxy
+
+#define DEV_CFG(dev)  ((const struct secproxy_mailbox_config *)((dev)->config))
+#define DEV_DATA(dev) ((struct secproxy_mailbox_data *)(dev)->data)
+
+#define DEV_TDATA(_dev) DEVICE_MMIO_NAMED_GET(_dev, target_data)
+#define DEV_RT(_dev)    DEVICE_MMIO_NAMED_GET(_dev, rt)
+#define DEV_SCFG(_dev)  DEVICE_MMIO_NAMED_GET(_dev, scfg)
+
+#define RT_THREAD_STATUS               0x0
+#define RT_THREAD_THRESHOLD            0x4
+#define RT_THREAD_STATUS_ERROR_SHIFT   31
+#define RT_THREAD_STATUS_ERROR_MASK    BIT(31)
+#define RT_THREAD_STATUS_CUR_CNT_SHIFT 0
+#define RT_THREAD_STATUS_CUR_CNT_MASK  GENMASK(7, 0)
+
+#define SCFG_THREAD_CTRL           0x1000
+#define SCFG_THREAD_CTRL_DIR_SHIFT 31
+#define SCFG_THREAD_CTRL_DIR_MASK  BIT(31)
+
+#define SEC_PROXY_THREAD(base, x) ((base) + (0x1000 * (x)))
+#define THREAD_IS_RX              1
+#define THREAD_IS_TX              0
+
+#define SECPROXY_MAILBOX_NUM_MSGS 5
+#define MAILBOX_MAX_CHANNELS      32
+#define MAILBOX_MBOX_SIZE         60
+
+#define SEC_PROXY_DATA_START_OFFS 0x4
+#define SEC_PROXY_DATA_END_OFFS   0x3c
+
+#define GET_MSG_SEQ(buffer) ((uint32_t *)buffer)[1]
+struct secproxy_thread {
+	mem_addr_t target_data;
+	mem_addr_t rt;
+	mem_addr_t scfg;
+};
+
+struct secproxy_mailbox_data {
+	mbox_callback_t cb[MAILBOX_MAX_CHANNELS];
+	void *user_data[MAILBOX_MAX_CHANNELS];
+	bool channel_enable[MAILBOX_MAX_CHANNELS];
+
+	DEVICE_MMIO_NAMED_RAM(target_data);
+	DEVICE_MMIO_NAMED_RAM(rt);
+	DEVICE_MMIO_NAMED_RAM(scfg);
+	struct k_spinlock lock;
+};
+
+struct secproxy_mailbox_config {
+	DEVICE_MMIO_NAMED_ROM(target_data);
+	DEVICE_MMIO_NAMED_ROM(rt);
+	DEVICE_MMIO_NAMED_ROM(scfg);
+	uint32_t irq;
+};
+
+static inline int secproxy_verify_thread(struct secproxy_thread *spt, uint8_t dir)
+{
+	/* Check for any errors already available */
+	if (sys_read32(spt->rt + RT_THREAD_STATUS) & RT_THREAD_STATUS_ERROR_MASK) {
+		LOG_ERR("Thread is corrupted, cannot send data.\n");
+		return -EINVAL;
+	}
```

Comment:
Have you looked at the equivalent driver in TF-A? `drivers/ti/ipc/sec_proxy.c` 
I see you're missing 2 things from there:
* Make sure thread is configured for right direction
* Check the message queue before sending/receiving data